src/main/java/org/terasology/web/resources/ResourceManager.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.web.resources;
+
+import org.terasology.context.Context;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.event.internal.EventSystem;
+import org.terasology.entitySystem.systems.ComponentSystem;
+import org.terasology.logic.console.Console;
+import org.terasology.network.NetworkSystem;
+import org.terasology.web.io.ActionResult;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class ResourceManager {
+
+    private static final ResourceManager INSTANCE = new ResourceManager();
+    private Map<String, Resource> resources;
+
+    public static ResourceManager getInstance() {
+        return INSTANCE;
+    }
+
+    public void initialize(Context context) {
+        if (resources == null) {
+            resources = new HashMap<>();
+            putResource(new ConsoleResource(context.get(Console.class)));
+            putResource(new OnlinePlayersResource(context.get(NetworkSystem.class)));
+        }
+        registerEventHandlers(context);
+    }
+
+    private void registerEventHandlers(Context context) {
+        EventSystem eventSystem = context.get(EntityManager.class).getEventSystem();
+        for (Map.Entry<String, Resource> entry: resources.entrySet()) {
+            Resource resource = entry.getValue();
+            if (resource instanceof ComponentSystem) {
```

Comment:
As far as I understand, it's completely automatic only if they are in a Module. The ones in the engine repository (like for example `ChatSystem`) are registered because they are part of the `engine` module, however I can't understand who "decides" whether a class on the classpath of the running JVM is part of it; I tried but my classes in FacadeServer are not automatically registered. Maybe I should investigate further and find out how to have them automatically registered if possible.

I changed `ResourceManager` (commit 	4a77c50) to use `ComponentSystemManager` to *almost* automatically subscribe to events and inject required context entries in each resource which is a `ComponentSystem`. I think it's a bit better than before since now I'm not directly interacting with `EventSystem` and `InjectionHelper` but delegating the work to `ComponentSystemManager` which in my understanding is more high-level and is designed exactly for this purpose. However it still isn't *fully* automatic as in Modules, is it okay for you?